starfish/experiment/builder/providers.py

```diff
@@ -49,6 +49,20 @@ def extras(self) -> dict:
         """
         return {}
 
+    @property
+    def filepath(self) -> str:
```

Comment:
```suggestion
    def filepath(self) -> Optional[str]:
```

Reply:
Hmm... I was trying to be consistent with the `extras` property. Should that also be changed to `Optional[dict]` and return `None`?